Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -105,43 +101,35 @@ WarpX::AddSpaceChargeFieldLabFrame ()
                                      "Error: RZ electrostatic only implemented for a single mode");
 #endif
 
-    // Allocate fields for charge
+    // Allocate fields for the potential
     // Also, zero out the phi data
     const int num_levels = max_level + 1;
-    Vector<std::unique_ptr<MultiFab> > rho(num_levels);
-    // Use number of guard cells used for local deposition of rho
-    const amrex::IntVect ng = guard_cells.ng_depos_rho;
     for (int lev = 0; lev <= max_level; lev++) {
         BoxArray nba = boxArray(lev);
         nba.surroundingNodes();
-        rho[lev] = std::make_unique<MultiFab>(nba, dmap[lev], 1, ng);
-        rho[lev]->setVal(0.);
         phi_fp[lev]->setVal(0.);
     }
 
     // Deposit particle charge density (source of Poisson solver)
     for (int ispecies=0; ispecies<mypc->nSpecies(); ispecies++){
         WarpXParticleContainer& species = mypc->GetParticleContainer(ispecies);
-        bool const local = true;
         bool const reset = false;
```

Comment:
Should reset = true, since you removed the line above setting rho to zero?

Reply:
Good point. I now set `rho_fp` to zero again.